group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,416 @@
+        // Consumer A
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, new HashSet<>(Arrays.asList(0, 1, 2)));
+        currentAssignmentForA.put(topic2Uuid, new HashSet<>(Collections.singletonList(0)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsA = new ArrayList<>(Collections.singletonList(topic1Uuid));
+        members.computeIfAbsent(consumerA, k -> new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsA, currentAssignmentForA));
+        // Consumer B
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic2Uuid, new HashSet<>(Collections.singletonList(1)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsB = new ArrayList<>(Arrays.asList(topic1Uuid, topic2Uuid, topic3Uuid));
+        members.computeIfAbsent(consumerB, k -> new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsB, currentAssignmentForB));
+        // Consumer C
+        Map<Uuid, Set<Integer>> currentAssignmentForC = new HashMap<>();
+        currentAssignmentForC.put(topic2Uuid, new HashSet<>(Collections.singletonList(2)));
+        currentAssignmentForC.put(topic3Uuid, new HashSet<>(Arrays.asList(0, 1)));
+        // Change subscriptions
+        List<Uuid> subscribedTopicsC = new ArrayList<>(Collections.singletonList(topic2Uuid));
+        members.put(consumerC, new AssignmentMemberSpec(Optional.empty(), Optional.empty(), subscribedTopicsC, new HashMap<>()));
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic1Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+        expectedAssignment.computeIfAbsent(topic1Uuid, k -> new HashSet<>()).add(new HashSet<>(Collections.singletonList(2)));
+        // Topic 2 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic2Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+        expectedAssignment.computeIfAbsent(topic2Uuid, k -> new HashSet<>()).add(new HashSet<>(Collections.singletonList(2)));
+        // Topic 3 Partitions Assignment
+        expectedAssignment.computeIfAbsent(topic3Uuid, k -> new HashSet<>()).add(new HashSet<>(Arrays.asList(0, 1)));
+
+        // Test for stickiness
+        assertTrue(computedAssignment.members().get(consumerC).targetPartitions().get(topic2Uuid).contains(2), "Stickiness test failed for Consumer C");
+        assertTrue(computedAssignment.members().get(consumerA).targetPartitions().get(topic1Uuid).containsAll(Arrays.asList(0, 1)), "Stickiness test failed for Consumer A");
+
+        assertAssignment(expectedAssignment, computedAssignment);
```

Comment:
we discussed this before when I was writing the tests, that is how I did it earlier but I realized that the order of members isn't guaranteed in the hashmap so we don't know exactly what order the assignor used to predict the expected assignment

Reply:
If two Maps have the same content, they will be equal, no? The order of the members does not matter here.

Reply:
the order in which they were assigned partitions is unknown to us since at the time of computing the coordinator accesses the members map (order not guaranteed) so we can't predict which member gets which partitions in the first place.  It could result in flaky tests